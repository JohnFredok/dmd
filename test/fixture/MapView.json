[
  {
    "id": "module:MapView",
    "name": "MapView",
    "kind": "module",
    "description": "MapView encapsulates a Google maps view so it can be used with famo.us.\n\nAdditionally it adds methods to set the position and zoom-factor of the map using transitions.\nUse `MapModifier` and `MapStateModifier` to place famo.us renderables on the map, much like google-maps markers.\n\n**Map-types**\n\n|Value|Description|\n|---|---|\n|MapType.GOOGLEMAPS (default)|Google-maps.|\n|MapType.LEAFLET|Leaflet.js.|"
  },
  {
    "id": "module:MapView^MapView",
    "name": "MapView",
    "kind": "class",
    "isExported": true,
    "alias": "module:MapView",
    "codeName": "MapView",
    "memberof": "module:MapView"
  },
  {
    "id": "module:MapView^MapViewâ—Š",
    "name": "MapView",
    "kind": "constructor",
    "memberof": "module:MapView^MapView",
    "params": [
      {
        "type": {
          "names": [
            "Object"
          ]
        },
        "description": "Options.",
        "name": "options"
      },
      {
        "type": {
          "names": [
            "MapType"
          ]
        },
        "description": "Map-type (e.g. MapView.MapType.GOOGLEMAPS, MapView.MapType.LEAFLET).",
        "name": "options.type"
      },
      {
        "type": {
          "names": [
            "Object"
          ]
        },
        "description": "Options that are passed directly to the Map object. The options should include the 'center' and 'zoom'.",
        "name": "options.mapOptions"
      },
      {
        "type": {
          "names": [
            "String"
          ]
        },
        "optional": true,
        "description": "Id of the DOM-element to use. When ommitted, a DOM-element is created using a surface.",
        "name": "options.id"
      },
      {
        "type": {
          "names": [
            "Transition"
          ]
        },
        "optional": true,
        "description": "Transition to use for smoothly zooming renderables (by default a transition of 120 ms is used).",
        "name": "options.zoomTransition"
      }
    ]
  },
  {
    "id": "module:MapView^MapView.DEFAULT_OPTIONS",
    "name": "DEFAULT_OPTIONS",
    "scope": "static",
    "kind": "member",
    "memberof": "module:MapView^MapView",
    "properties": [
      {
        "name": "DEFAULT_OPTIONS"
      }
    ],
    "access": "protected",
    "codeName": "MapView.DEFAULT_OPTIONS"
  },
  {
    "id": "module:MapView^MapView#getMap",
    "name": "getMap",
    "scope": "instance",
    "kind": "function",
    "description": "Get the internal map-object. This object may not yet have been initialized, the map is only\nguarenteed to be valid after the 'load' event has been emited.",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "Map"
          ]
        },
        "description": "Map object."
      }
    ],
    "codeName": "MapView.prototype.getMap"
  },
  {
    "id": "module:MapView^MapView#setPosition",
    "name": "setPosition",
    "scope": "instance",
    "kind": "function",
    "description": "Set the center of the map to the given geographical coordinates.",
    "memberof": "module:MapView^MapView",
    "params": [
      {
        "type": {
          "names": [
            "LatLng"
          ]
        },
        "description": "Position in geographical coordinates.",
        "name": "position"
      },
      {
        "type": {
          "names": [
            "Transitionable"
          ]
        },
        "optional": true,
        "description": "Transitionable.",
        "name": "transition"
      },
      {
        "type": {
          "names": [
            "function"
          ]
        },
        "optional": true,
        "description": "callback to call after transition completes.",
        "name": "callback"
      }
    ],
    "codeName": "MapView.prototype.setPosition"
  },
  {
    "id": "module:MapView^MapView#getPosition",
    "name": "getPosition",
    "scope": "instance",
    "kind": "function",
    "description": "Get the current center position of the map, in geographical coordinates.",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "LatLng"
          ]
        },
        "description": "Position in geographical coordinates."
      }
    ],
    "codeName": "MapView.prototype.getPosition"
  },
  {
    "id": "module:MapView^MapView#getFinalPosition",
    "name": "getFinalPosition",
    "scope": "instance",
    "kind": "function",
    "description": "Get the destination center position of the map, in geographical coordinates.",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "LatLng"
          ]
        },
        "description": "Position in geographical coordinates."
      }
    ],
    "codeName": "MapView.prototype.getFinalPosition"
  },
  {
    "id": "module:MapView^MapView#getZoom",
    "name": "getZoom",
    "scope": "instance",
    "kind": "function",
    "description": "Get the current zoom-level of the map, taking into account smooth transition between zoom-levels.\nE.g., when zooming from zoom-level 4 to 5, this function returns an increasing value starting at 4 and ending\nat 5, over time. The used zoomTransition can be set as an option.",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "Number"
          ]
        },
        "description": "Zoom-level."
      }
    ],
    "codeName": "MapView.prototype.getZoom"
  },
  {
    "id": "module:MapView^MapView#pointFromPosition",
    "name": "pointFromPosition",
    "scope": "instance",
    "kind": "function",
    "description": "Get the position in pixels (relative to the left-top of the container) for the given geographical position.",
    "memberof": "module:MapView^MapView",
    "params": [
      {
        "type": {
          "names": [
            "LatLng"
          ]
        },
        "description": "in geographical coordinates.",
        "name": "position"
      }
    ],
    "returns": [
      {
        "type": {
          "names": [
            "Point"
          ]
        },
        "description": "Position in pixels, relative to the left-top of the mapView."
      }
    ],
    "codeName": "MapView.prototype.pointFromPosition"
  },
  {
    "id": "module:MapView^MapView#positionFromPoint",
    "name": "positionFromPoint",
    "scope": "instance",
    "kind": "function",
    "description": "Get the geographical coordinates for a given position in pixels (relative to the left-top of the container).",
    "memberof": "module:MapView^MapView",
    "params": [
      {
        "type": {
          "names": [
            "Point"
          ]
        },
        "description": "Position in pixels, relative to the left-top of the mapView.",
        "name": "point"
      }
    ],
    "returns": [
      {
        "type": {
          "names": [
            "LatLng"
          ]
        },
        "description": "Position in geographical coordinates."
      }
    ],
    "codeName": "MapView.prototype.positionFromPoint"
  },
  {
    "id": "module:MapView^MapView#getSize",
    "name": "getSize",
    "scope": "instance",
    "kind": "function",
    "description": "Get the size of the map-view in pixels.",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "Array.Number"
          ]
        },
        "description": "Size of the mapView."
      }
    ],
    "codeName": "MapView.prototype.getSize"
  },
  {
    "id": "module:MapView^MapView#halt",
    "name": "halt",
    "scope": "instance",
    "kind": "function",
    "description": "Halts any pending transitions.",
    "memberof": "module:MapView^MapView",
    "codeName": "MapView.prototype.halt"
  },
  {
    "id": "module:MapView^MapView#isActive",
    "name": "isActive",
    "scope": "instance",
    "kind": "function",
    "description": "Is there at least one action pending completion?",
    "memberof": "module:MapView^MapView",
    "returns": [
      {
        "type": {
          "names": [
            "Bool"
          ]
        },
        "description": "True when there are active transitions running."
      }
    ],
    "codeName": "MapView.prototype.isActive"
  }
]