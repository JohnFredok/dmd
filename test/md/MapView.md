<a name="module_MapView"></a>
##MapView
MapView encapsulates a Google maps view so it can be used with famo.us.

Additionally it adds methods to set the position and zoom-factor of the map using transitions.
Use `MapModifier` and `MapStateModifier` to place famo.us renderables on the map, much like google-maps markers.

**Map-types**

|Value|Description|
|---|---|
|MapType.GOOGLEMAPS (default)|Google-maps.|
|MapType.LEAFLET|Leaflet.js.|

* [MapView](#module_MapView)
  * [class: MapView ⏏](#exp_module_MapView^MapView)
    * [`new MapView(options)`](#new_module_MapView^MapView())
    * [`MapView.DEFAULT_OPTIONS`](#module_MapView^MapView.DEFAULT_OPTIONS)
    * [`mapView.getMap()` ⇒ `Map`](#module_MapView^MapView#getMap)
    * [`mapView.setPosition(position, [transition], [callback])`](#module_MapView^MapView#setPosition)
    * [`mapView.getPosition()` ⇒ `LatLng`](#module_MapView^MapView#getPosition)
    * [`mapView.getFinalPosition()` ⇒ `LatLng`](#module_MapView^MapView#getFinalPosition)
    * [`mapView.getZoom()` ⇒ `Number`](#module_MapView^MapView#getZoom)
    * [`mapView.pointFromPosition(position)` ⇒ `Point`](#module_MapView^MapView#pointFromPosition)
    * [`mapView.positionFromPoint(point)` ⇒ `LatLng`](#module_MapView^MapView#positionFromPoint)
    * [`mapView.getSize()` ⇒ `Array.Number`](#module_MapView^MapView#getSize)
    * [`mapView.halt()`](#module_MapView^MapView#halt)
    * [`mapView.isActive()` ⇒ `Bool`](#module_MapView^MapView#isActive)

<a name="exp_module_MapView^MapView"></a>
###class: MapView ⏏
<a name="new_module_MapView^MapView()"></a>
####`new MapView(options)`
| Param | Type | Description |
| ----- | ---- | ----------- |
| options | `Object` | Options. |
| options.type | `MapType` | Map-type (e.g. MapView.MapType.GOOGLEMAPS, MapView.MapType.LEAFLET). |
| options.mapOptions | `Object` | Options that are passed directly to the Map object. The options should include the 'center' and 'zoom'. |
| options.id | `String` | Id of the DOM-element to use. When ommitted, a DOM-element is created using a surface. |
| options.zoomTransition | `Transition` | Transition to use for smoothly zooming renderables (by default a transition of 120 ms is used). |
<a name="module_MapView^MapView.DEFAULT_OPTIONS"></a>
####`MapView.DEFAULT_OPTIONS`
**Access**: protected  
<a name="module_MapView^MapView#getMap"></a>
####`mapView.getMap()` ⇒ `Map`
Get the internal map-object. This object may not yet have been initialized, the map is only
guarenteed to be valid after the 'load' event has been emited.

**Returns**: `Map` - Map object.  
<a name="module_MapView^MapView#setPosition"></a>
####`mapView.setPosition(position, [transition], [callback])`
Set the center of the map to the given geographical coordinates.

| Param | Type | Description |
| ----- | ---- | ----------- |
| position | `LatLng` | Position in geographical coordinates. |
| transition | `Transitionable` | Transitionable. |
| callback | `function` | callback to call after transition completes. |
<a name="module_MapView^MapView#getPosition"></a>
####`mapView.getPosition()` ⇒ `LatLng`
Get the current center position of the map, in geographical coordinates.

**Returns**: `LatLng` - Position in geographical coordinates.  
<a name="module_MapView^MapView#getFinalPosition"></a>
####`mapView.getFinalPosition()` ⇒ `LatLng`
Get the destination center position of the map, in geographical coordinates.

**Returns**: `LatLng` - Position in geographical coordinates.  
<a name="module_MapView^MapView#getZoom"></a>
####`mapView.getZoom()` ⇒ `Number`
Get the current zoom-level of the map, taking into account smooth transition between zoom-levels.
E.g., when zooming from zoom-level 4 to 5, this function returns an increasing value starting at 4 and ending
at 5, over time. The used zoomTransition can be set as an option.

**Returns**: `Number` - Zoom-level.  
<a name="module_MapView^MapView#pointFromPosition"></a>
####`mapView.pointFromPosition(position)` ⇒ `Point`
Get the position in pixels (relative to the left-top of the container) for the given geographical position.

| Param | Type | Description |
| ----- | ---- | ----------- |
| position | `LatLng` | in geographical coordinates. |
**Returns**: `Point` - Position in pixels, relative to the left-top of the mapView.  
<a name="module_MapView^MapView#positionFromPoint"></a>
####`mapView.positionFromPoint(point)` ⇒ `LatLng`
Get the geographical coordinates for a given position in pixels (relative to the left-top of the container).

| Param | Type | Description |
| ----- | ---- | ----------- |
| point | `Point` | Position in pixels, relative to the left-top of the mapView. |
**Returns**: `LatLng` - Position in geographical coordinates.  
<a name="module_MapView^MapView#getSize"></a>
####`mapView.getSize()` ⇒ `Array.Number`
Get the size of the map-view in pixels.

**Returns**: `Array.Number` - Size of the mapView.  
<a name="module_MapView^MapView#halt"></a>
####`mapView.halt()`
Halts any pending transitions.

<a name="module_MapView^MapView#isActive"></a>
####`mapView.isActive()` ⇒ `Bool`
Is there at least one action pending completion?

**Returns**: `Bool` - True when there are active transitions running.  
