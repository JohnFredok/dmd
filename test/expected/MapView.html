<!DOCTYPE html>
<html>
  <head>
    <title>Docs</title>
    <meta charset="utf8">
    <link rel="stylesheet" href="/html/docs.css">
    <link rel="stylesheet" href="/html/coy.css">
  </head>
  <body class="overflow-hidden">
    <div class="flexbox-col box-fill-viewport">
      <div class="row1 flex-fixed">
        <h1>API Reference</h1>
      </div>
      <div class="row2 flexbox-row">
        <div class="col1 horiz-space flex-fixed overflow-scroll">
          <h3>Modules</h3><ul><li>
  <a href="#module_MapView">MapView</a>
  
    <ul>
    <li>
  <a href="#exp_module_MapView^MapView">class: MapView ⏏</a>
  
    <ul>
    <li>
  <a href="#new_module_MapView^MapView◊">new MapView(options)</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#isActive">mapView.isActive()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#getMap">mapView.getMap()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#setPosition">mapView.setPosition(position, [transition], [callback])</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#getPosition">mapView.getPosition()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#getFinalPosition">mapView.getFinalPosition()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#halt">mapView.halt()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#pointFromPosition">mapView.pointFromPosition(position)</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#positionFromPoint">mapView.positionFromPoint(point)</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#getSize">mapView.getSize()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView#getZoom">mapView.getZoom()</a>
  
</li>

  
    
    <li>
  <a href="#module_MapView^MapView.DEFAULT_OPTIONS">MapView.DEFAULT_OPTIONS</a>
  
</li>
</ul>
  
</li>
</ul>
  
</li>
</ul>
          

          

          
        </div>
        <div class="col2 overflow-scroll flex-full">
        
          <h2><a name="module_MapView">MapView</a></h2>
<p>MapView encapsulates a Google maps view so it can be used with famo.us.</p>
<p>Additionally it adds methods to set the position and zoom-factor of the map using transitions.
Use <code>MapModifier</code> and <code>MapStateModifier</code> to place famo.us renderables on the map, much like google-maps markers.</p>
<p><strong>Map-types</strong></p>
<table>
<thead>
<tr>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>MapType.GOOGLEMAPS (default)</td>
<td>Google-maps.</td>
</tr>
<tr>
<td>MapType.LEAFLET</td>
<td>Leaflet.js.</td>
</tr>
</tbody>
</table>






<dl>
  <dt>kind</dt><dd>module</dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h3><a name="exp_module_MapView^MapView">class: MapView ⏏</a></h3>






<dl>
  <dt>kind</dt><dd>class</dd>
  
  <dt>memberof</dt><dd><code> class of <a href="#module_MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="new_module_MapView^MapView◊">new MapView(options)</a></h4>


<strong>Params</strong>
<ul>
<li>options <code>Object</code> Options.</li>

<li>options.type <code>MapType</code> Map-type (e.g. MapView.MapType.GOOGLEMAPS, MapView.MapType.LEAFLET).</li>

<li>options.mapOptions <code>Object</code> Options that are passed directly to the Map object. The options should include the &#39;center&#39; and &#39;zoom&#39;.</li>

<li>options.id <code>String</code> Id of the DOM-element to use. When ommitted, a DOM-element is created using a surface.</li>

<li>options.zoomTransition <code>Transition</code> Transition to use for smoothly zooming renderables (by default a transition of 120 ms is used).</li>
</ul>




<dl>
  <dt>kind</dt><dd>constructor</dd>
  
  <dt>memberof</dt><dd><code> constructor of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#isActive">mapView.isActive() -> `Bool`</a></h4>
<p>Is there at least one action pending completion?</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#getMap">mapView.getMap() -> `Map`</a></h4>
<p>Get the internal map-object. This object may not yet have been initialized, the map is only
guarenteed to be valid after the &#39;load&#39; event has been emited.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#setPosition">mapView.setPosition(position, [transition], [callback])</a></h4>
<p>Set the center of the map to the given geographical coordinates.</p>


<strong>Params</strong>
<ul>
<li>position <code>LatLng</code> Position in geographical coordinates.</li>

<li>transition <code>Transitionable</code> Transitionable.</li>

<li>callback <code>function</code> callback to call after transition completes.</li>
</ul>




<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#getPosition">mapView.getPosition() -> `LatLng`</a></h4>
<p>Get the current center position of the map, in geographical coordinates.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#getFinalPosition">mapView.getFinalPosition() -> `LatLng`</a></h4>
<p>Get the destination center position of the map, in geographical coordinates.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#halt">mapView.halt()</a></h4>
<p>Halts any pending transitions.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#pointFromPosition">mapView.pointFromPosition(position) -> `Point`</a></h4>
<p>Get the position in pixels (relative to the left-top of the container) for the given geographical position.</p>


<strong>Params</strong>
<ul>
<li>position <code>LatLng</code> in geographical coordinates.</li>
</ul>




<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#positionFromPoint">mapView.positionFromPoint(point) -> `LatLng`</a></h4>
<p>Get the geographical coordinates for a given position in pixels (relative to the left-top of the container).</p>


<strong>Params</strong>
<ul>
<li>point <code>Point</code> Position in pixels, relative to the left-top of the mapView.</li>
</ul>




<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#getSize">mapView.getSize() -> `Array.Number`</a></h4>
<p>Get the size of the map-view in pixels.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView#getZoom">mapView.getZoom() -> `Number`</a></h4>
<p>Get the current zoom-level of the map, taking into account smooth transition between zoom-levels.
E.g., when zooming from zoom-level 4 to 5, this function returns an increasing value starting at 4 and ending
at 5, over time. The used zoomTransition can be set as an option.</p>






<dl>
  <dt>kind</dt><dd>function</dd>
  <dt>scope</dt><dd>instance</dd>
  <dt>memberof</dt><dd><code>instance function of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
</dl>

<h4><a name="module_MapView^MapView.DEFAULT_OPTIONS">MapView.DEFAULT_OPTIONS</a></h4>






<dl>
  <dt>kind</dt><dd>member</dd>
  <dt>scope</dt><dd>static</dd>
  <dt>memberof</dt><dd><code>static member of <a href="#exp_module_MapView^MapView">MapView</a></code></dd>
  
  
  
  
  
  
  
  
  
  <dt>access</dt><dd>protected</dd>
  
  
  
  
  
  
</dl>


        </div>
      </div>
    </div>
    <script src="/html/prism.js"></script>
  </body>
</html>
